torchgeo/datasets/mmflood.py

```diff
@@ -0,0 +1,382 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""MMFlood dataset."""
+
+from __future__ import annotations
+
+import os
+from collections.abc import Callable
+from glob import glob
+from typing import ClassVar, Literal
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from matplotlib.figure import Figure
+from rasterio.crs import CRS
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import IntersectionDataset, RasterDataset
+from .utils import BoundingBox, Path, download_url, extract_archive
+
+
+class MMFloodComponent(RasterDataset):
+    """Base component for MMFlood dataset."""
+
+    def __init__(
+        self,
+        subfolders: list[str],
+        content: Literal['s1_raw', 'DEM', 'hydro', 'mask'],
+        root: Path = 'data',
+        crs: CRS | None = None,
+        res: float | None = None,
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        cache: bool = False,
+    ) -> None:
+        """Initialize MMFloodComponent dataset instance.
+
+        Args:
+            subfolders: list of directories to be loaded
+            content: specifies which component to load
+            root: root directory where dataset can be found
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+                (defaults to the resolution of the first file found)
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+        """
+        self.content = content
+        self.is_image = content != 'mask'
+        paths = []
+        for s in subfolders:
+            paths += glob(os.path.join(root, '**', f'{s}*-*', self.content, '*.tif'))
+        paths = sorted(paths)
+        super().__init__(paths, crs, res, transforms=transforms, cache=cache)
+
+
+class MMFloodIntersection(IntersectionDataset):
+    """Intersection dataset used to merge two or more MMFloodComponents."""
```

Comment:
Do these helper classes also need a versionadded? @adamjstewart

Reply:
Not really, they won't appear in the documentation anyway.